### PR TITLE
[PR2] DEV-70 add PacketParser

### PR DESCRIPTION
**Pull request by jmaralo**
The PacketParser is in charge of encoding and decoding packets based on the information it's provided.

There are various encoders and decoders for each type of value. The code is almost the same as the previous PacketParser, but it adds a method (AddPacket) which is used by the ExcelAdapter to build the struct from the excel.

### CODE REVIEW

File: src/packet_parser/packet_parser.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/HyperloopUPV-H8/Backend-H8/packet_parser/models"
+)
+
+type PacketParser struct {
+	descriptors map[uint16]models.PacketDescriptor
+	enums       map[string]models.Enum
+}
+
+func (parser *PacketParser) AddPacket(board string, ip string, desc excelAdapterModels.Description, values []excelAdapterModels.Value) {
+	if parser.descriptors == nil {
```
Comment (smorsua) [inline src/packet_parser/packet_parser.go:21]: Possible error should be handled outside this function.

File: src/packet_parser/packet_parser.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+		log.Fatalf("packet parser: AddPacket: %s\n", err)
+	}
+
+	valueDescriptors := make([]models.ValueDescriptor, len(values))
+	for i, value := range values {
+		kind := value.Type
+		if strings.HasPrefix(kind, "ENUM") {
+			kind = "enum"
+			if parser.enums == nil {
+				parser.enums = make(map[string]models.Enum)
```
Comment (smorsua) [inline src/packet_parser/packet_parser.go:36]: Same with this

**Review by smorsua — CHANGES_REQUESTED** [review]
Some functions have responsability that doesnt belong to them. Looks good overall anyways

File: src/packet_parser/packet_parser.go (hunk 1 above)
Comment (lipezaballa) [reply src/packet_parser/packet_parser.go:21]: and where should it be?

**Review by lipezaballa — APPROVED** [review]

**Review by smorsua — APPROVED** [review]

**Pull request merged**